sdk/src/Services/S3/Custom/Transfer/Internal/MultipartUploadCommand.cs

```diff
@@ -230,7 +247,43 @@ private UploadPartRequest ConstructUploadPartRequest(int partNumber, long filePo
             return uploadRequest;
         }
 
+        private UploadPartRequest ConstructUploadPartRequestForNonSeekableStream(Stream inputStream, int partNumber, long partSize, bool isLastPart, InitiateMultipartUploadResponse initiateResponse)
+        {
+            UploadPartRequest uploadPartRequest = new UploadPartRequest()
```

Comment:
It would be good to refactor this a bit more so there is a method to setup the generic `UploadPartRequest` that also sets the SigV4 and then have the callers just add the settings that are specific to either streaming or not. We know we keep adding properties to the S3 upload and it seems like these methods will go out of sync.

Reply:
Fixed. Please review.